src/extensions/replay/config.ts

```diff
@@ -114,17 +132,46 @@ const limitPayloadSize = (
 
     return (data) => {
         if (data?.requestBody) {
-            data.requestBody = redactPayload(data.requestBody, data.requestHeaders, limit, 'Request')
+            data.requestBody = enforcePayloadSizeLimit(data.requestBody, data.requestHeaders, limit, 'Request')
         }
 
         if (data?.responseBody) {
-            data.responseBody = redactPayload(data.responseBody, data.responseHeaders, limit, 'Response')
+            data.responseBody = enforcePayloadSizeLimit(data.responseBody, data.responseHeaders, limit, 'Response')
         }
 
         return data
     }
 }
 
+function scrubPayload(payload: string | null | undefined, label: 'Request' | 'Response'): string | null | undefined {
+    if (_isNullish(payload)) {
+        return payload
+    }
+    let scrubbed = payload
+
+    if (!shouldCaptureValue(scrubbed, false)) {
+        scrubbed = LOGGER_PREFIX + ' ' + label + ' body redacted'
+    }
+    _each(PAYLOAD_CONTENT_DENY_LIST, (text) => {
+        if (scrubbed?.length && scrubbed?.indexOf(text) !== -1) {
+            scrubbed = LOGGER_PREFIX + ' ' + label + ' body might contain: ' + text
+        }
+    })
+
+    return scrubbed
+}
+
+function scrubPayloads(capturedRequest: CapturedNetworkRequest | undefined): CapturedNetworkRequest | undefined {
```

Comment:
Probably worth scrubbing the URL as well.
Eg google.com/login/password/123 or google.com/login?password=123 will be captured and there's a PII risk

Reply:
no need to block this PR but worth fixing as part of the TODOs (ps I was part of incidents with URL paths and query strings before), so definitely a must.